### PR TITLE
Re-organize operators into separate files

### PR DESCRIPTION
**Pull request by tonyd256**
This moves the monadic operators into their own files and renames the
files with the Argo operators to (ReturnType)Pull.swift.

### CODE REVIEW

**Comment by tonyd256** [comment]
An attempt at #253 and #254

**Comment by gfontenot** [comment]
I don't think this actually solves #254, right? Am I missing it somewhere?

I'm good with this. I think it's the most reasonable change. However, I question the use of `Pull` as the name for the file containing the `<|` family of operators. I don't _believe_ we've been using the word `pull` for that operator. I thought we had generally settled on `decode` as the word. I'd also probably flip the words so that it became `DecodeOptional` and `DecodeDecoded` (or whatever we actually settle on for the name of `<|`). Seems like they read better, and they'd sort together alphabetically as well.

**Comment by tonyd256** [comment]
That's fine with me. I actually had it that way initially then changed when I read "Pull" in the comments.

**Comment by tonyd256** [comment]
It solves #254 in that it moves `pure` into the `Applicative` file since it is a function on Applicative in Haskell.

**Comment by gfontenot** [comment]
I don't see `pure` in `Applicative.swift`

**Comment by tonyd256** [comment]
@gfontenot yup sorry. I thought I had done it. It's done now.

**Comment by gfontenot** [comment]
dig it

Maybe taking things too far, but: what do you think about moving the implementations of `map`, `apply`, `flatMap` into `Functor.swift`, `Applicative.swift`, and `Monad.swift`? We already have a non-operator in there, and if those are about type conformance, maybe it makes sense? We could also move them out of an `Operators` directory. Maybe `Argo/Decoded/Monad.swift` and `Argo/Decoded/Decoded.swift`?

**Comment by gfontenot** [comment]
if that's insane, feel free to merge because this is a huge step up as is.

**Comment by tonyd256** [comment]
I actually like that idea! Will do! Then will merge!

**Comment by tonyd256** [comment]
@gfontenot Done. Didn't merge yet. Want to make sure it looks good.

File: Argo/Types/Decoded/Decoded.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ public extension Decoded {
 }
 
 extension Decoded: CustomStringConvertible {
-  public var description: String {
+   public var description: String {
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: ugh sorry

**Comment by gfontenot** [comment]
I think this is a great change. Let's merge this shit.

**Pull request merged**